applications/serial_lte_modem/src/slm_at_socket.c

```diff
@@ -0,0 +1,1579 @@
+		err = 0;
+		break;
+
+	case AT_CMD_TYPE_TEST_COMMAND:
+		sprintf(rsp_buf, "\r\n#XSSOCKET: (%d,%d,%d),(%d,%d),(%d,%d)",
+			AT_SOCKET_CLOSE, AT_SOCKET_OPEN, AT_SOCKET_OPEN6,
+			SOCK_STREAM, SOCK_DGRAM,
+			AT_SOCKET_ROLE_CLIENT, AT_SOCKET_ROLE_SERVER);
+		strcat(rsp_buf, "<sec-tag>,<peer_verify>\r\n");
+		rsp_send(rsp_buf, strlen(rsp_buf));
+		err = 0;
+		break;
+
+	default:
+		break;
+	}
+
+	return err;
+}
+
+/**@brief handle AT#XSOCKETOPT commands
+ *  AT#XSOCKETOPT=<op>,<name>[,<value>]
+ *  AT#XSOCKETOPT? READ command not supported
+ *  AT#XSOCKETOPT=?
+ */
+int handle_at_socketopt(enum at_cmd_type cmd_type)
+{
+	int err = -EINVAL;
+	uint16_t op;
+	uint16_t name;
+	enum at_param_type type = AT_PARAM_TYPE_NUM_INT;
+
+	switch (cmd_type) {
+	case AT_CMD_TYPE_SET_COMMAND:
+		if (sock.fd < 0) {
+			LOG_ERR("Socket not opened yet");
+			return err;
+		}
+		err = at_params_unsigned_short_get(&at_param_list, 1, &op);
+		if (err) {
+			return err;
+		}
+		err = at_params_unsigned_short_get(&at_param_list, 2, &name);
+		if (err) {
+			return err;
+		}
+		if (op == AT_SOCKETOPT_SET) {
+			int value_int = 0;
+			char value_str[IFNAMSIZ] = {0};
+			int size = IFNAMSIZ;
+
+			if (at_params_valid_count_get(&at_param_list) > 3) {
+				type = at_params_type_get(&at_param_list, 3);
+				if (type == AT_PARAM_TYPE_NUM_INT) {
+					err = at_params_int_get(&at_param_list, 3, &value_int);
+					if (err) {
+						return err;
+					}
+				} else if (type == AT_PARAM_TYPE_STRING) {
+					err = util_string_get(&at_param_list, 3, value_str, &size);
+					if (err) {
+						return err;
+					}
+				} else {
+					return -EINVAL;
+				}
+			}
+			if (type == AT_PARAM_TYPE_NUM_INT) {
+				err = do_socketopt_set_int(name, value_int);
+			} else if (type == AT_PARAM_TYPE_STRING) {
+				err = do_socketopt_set_str(name, value_str);
+			}
+		} else if (op == AT_SOCKETOPT_GET) {
+			err = do_socketopt_get(name);
+		} break;
+
+	case AT_CMD_TYPE_TEST_COMMAND:
+		sprintf(rsp_buf, "\r\n#XSOCKETOPT: (%d,%d),<name>,<value>\r\n",
+			AT_SOCKETOPT_GET, AT_SOCKETOPT_SET);
+		rsp_send(rsp_buf, strlen(rsp_buf));
+		err = 0;
+		break;
+
+	default:
+		break;
+	}
+
+	return err;
+}
+
+/**@brief handle AT#XSSOCKETOPT commands
+ *  AT#XSSOCKETOPT=<op>,<name>[,<value>]
+ *  AT#XSSOCKETOPT? READ command not supported
+ *  AT#XSSOCKETOPT=?
+ */
+int handle_at_secure_socketopt(enum at_cmd_type cmd_type)
+{
+	int err = -EINVAL;
+	uint16_t op;
+	uint16_t name;
+	enum at_param_type type = AT_PARAM_TYPE_NUM_INT;
+
+	switch (cmd_type) {
+	case AT_CMD_TYPE_SET_COMMAND:
+		if (sock.fd < 0 || sock.sec_tag == INVALID_SEC_TAG) {
+			LOG_ERR("Secure socket not opened yet");
+			return err;
+		}
+		err = at_params_unsigned_short_get(&at_param_list, 1, &op);
+		if (err) {
+			return err;
+		}
+		err = at_params_unsigned_short_get(&at_param_list, 2, &name);
+		if (err) {
+			return err;
+		}
+		if (op == AT_SOCKETOPT_SET) {
+			int value_int = 0;
+			char value_str[IFNAMSIZ] = {0};
```

Comment:
I think that SLM_MAX_URL would be a better choice here, since the only secure option accepting string is TLS_HOSTNAME.

Reply:
Thanks, applied